exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/statistics/StatisticsMergeBatch.java

```diff
@@ -0,0 +1,532 @@
+          NullableFloat8Holder sumHolder;
+          String colName = mapElt.getField().getLastName();
+          if (statMap == null) {
+            statMap = aggregationMap.get(functions.get(vv.getField().getLastName()));
+          }
+          if (statMap.get(colName) != null) {
+            sumHolder = (NullableFloat8Holder) statMap.get(colName);
+          } else {
+            sumHolder = new NullableFloat8Holder();
+            statMap.put(colName, sumHolder);
+          }
+          //TODO: assert size = 1
+          //TODO: logger
+          Object val = mapElt.getAccessor().getObject(0);
+          if (val != null) {
+            sumHolder.value += (double) val;
+            sumHolder.isSet = 1;
+          }
+        } else if (vv.getField().getLastName().equalsIgnoreCase("hll")) {
+          ObjectHolder hllHolder;
+          String colName = mapElt.getField().getLastName();
+          if (statMap == null) {
+            statMap = aggregationMap.get(functions.get(vv.getField().getLastName()));
+          }
+          if (statMap.get(colName) != null) {
+            hllHolder = (ObjectHolder) statMap.get(colName);
+          } else {
+            hllHolder = new ObjectHolder();
+            hllHolder.obj = new HyperLogLog(context.getContextInformation().getHllAccuracy());
+            statMap.put(colName, hllHolder);
+          }
+          NullableVarBinaryVector hllVector = (NullableVarBinaryVector) mapElt;
+          try {
+            if (hllVector.getAccessor().isSet(0) == 1) {
+              ByteArrayInputStream bais = new ByteArrayInputStream(hllVector.getAccessor().getObject(0), 0,
+                  mapElt.getBufferSize());
+              HyperLogLog other = HyperLogLog.Builder.build(new DataInputStream(bais));
+              ((HyperLogLog) hllHolder.obj).addAll(other);
+            }
+          } catch (Exception ex) {
+            //TODO: Catch IOException/CardinalityMergeException
+            //TODO: logger
+            return IterOutcome.STOP;
+          }
+        }
+      }
+      // Add NDV value vector map using HLL map (since the NDV map is directly generated from HLL and not produced by the underlying
+      // Statistics Agg)
+      Map<String, ValueHolder> hllMap = aggregationMap.get("hll");
+      if (hllMap != null) {
+        aggregationMap.put("ndv", hllMap);
+      }
+    }
+    return IterOutcome.OK;
+  }
+
+  public VectorContainer getOutgoingContainer() {
+    return this.container;
+  }
+
+  @Override
+  public IterOutcome innerNext() {
+    IterOutcome outcome;
+    boolean didSomeWork = false;
+    if (finished) {
+      return IterOutcome.NONE;
+    }
+    try {
+      outer: while (true) {
+        outcome = next(incoming);
+        switch (outcome) {
+          case NONE:
+            break outer;
+          case OUT_OF_MEMORY:
+          case NOT_YET:
+          case STOP:
+            return outcome;
+          case OK_NEW_SCHEMA:
+            if (first) {
+              first =false;
```

Comment:
This is borrowed from the existing state management code in other operators. We should change them across the board rather than doing it for a few operators in a separate commit.